packages/starknet-snap/src/rpcs/estimateFee.test.ts

```diff
@@ -0,0 +1,132 @@
+import { InvalidParamsError } from '@metamask/snaps-sdk';
+import type { Invocations } from 'starknet';
+import { constants, TransactionType } from 'starknet';
+
+import { FeeTokenUnit } from '../types/snapApi';
+import { STARKNET_SEPOLIA_TESTNET_NETWORK } from '../utils/constants';
+import * as starknetUtils from '../utils/starknetUtils';
+import { mockAccount, prepareMockAccount } from './__tests__/helper';
+import { estimateFee } from './estimateFee';
+import type { EstimateFeeParams } from './estimateFee';
+
+jest.mock('../utils/snap');
+jest.mock('../utils/logger');
+jest.mock('../utils/starknetUtils');
+
+describe('estimateFee', () => {
+  const state = {
+    accContracts: [],
+    erc20Tokens: [],
+    networks: [STARKNET_SEPOLIA_TESTNET_NETWORK],
+    transactions: [],
+  };
+
+  it('estimates fee correctly when account is deployed', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+
+    prepareMockAccount(account, state);
+
+    const estimateBulkFeeRespMock = {
+      suggestedMaxFee: BigInt(1000000000000000).toString(10),
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      overallFee: BigInt(1500000000000000).toString(10),
+      unit: FeeTokenUnit.ETH,
+      includeDeploy: false,
+    };
+
+    const getEstimatedFees = jest.spyOn(starknetUtils, 'getEstimatedFees');
+    getEstimatedFees.mockResolvedValue(estimateBulkFeeRespMock);
+
+    (starknetUtils.isAccountDeployed as jest.Mock).mockResolvedValue(true);
+
+    (starknetUtils.estimateFeeBulk as jest.Mock).mockResolvedValue(
+      estimateBulkFeeRespMock,
+    );
+    (starknetUtils.addFeesFromAllTransactions as jest.Mock).mockReturnValue(
+      estimateBulkFeeRespMock,
+    );
```

Comment:
create an helper function on this file? such as prepareMock, and add some default value, 
and you dont need to mock addFeesFromAllTransactions